### PR TITLE
Add oaidl::ITypeLib, oleauto::LoadTypeLibEx and related definitions.

### PR DESCRIPTION
**Pull request by Arnavion**
There's more in oleauto.h but I've only added what I need for winapi-tlb-bindgen.

There's some amount of duplication between oaidl.h and oleauto.h (both define `DISPID` and `MEMBERID`, etc) but I figured it would be better to reuse the definitions from the `oaidl` module.

`MEMBERID_NIL`'s type comes from its use as the first parameter of [`ITypeInfo::GetDocumentation()`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms221227)

### CODE REVIEW

File: src/um/oleauto.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+use um::oaidl::{DISPID_UNKNOWN, ITypeLib, MEMBERID};
+use um::winnt::{HRESULT, INT};
+
+EXTERN!{stdcall fn SysAllocString(
+    psz: *const OLECHAR
+) -> BSTR}
+
+EXTERN!{stdcall fn SysFreeString(
+    bstrString: BSTR
+) -> ()}
```
Comment (Arnavion) [inline src/um/oleauto.rs]: Is this correct or should it be c_void or something?

Comment (retep998) [inline src/um/oleauto.rs]: `void` return types correspond to `()` in Rust.

File: src/um/oaidl.rs [hunk 2]
```diff
@@ -526,7 +526,53 @@ interface IDispatch(IDispatchVtbl): IUnknown(IUnknownVtbl) {
 );
 pub enum IRecordInfo {} // FIXME
 pub enum ITypeComp {} // FIXME
-pub enum ITypeLib {} // FIXME
+
+ENUM! { enum SYSKIND {
```
Comment (retep998) [inline src/um/oaidl.rs]: Incorrect spacing.

File: src/um/oleauto.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Copyright © 2016-2017 winapi-rs developers
```
Comment (retep998) [inline src/um/oleauto.rs]: Shouldn't this be just 2017?

**Comment by retep998** [comment]
If there is duplication the definitions should be duplicated except for structs and interfaces, where one location should be canonical and then the other headers publicly re-export it.

**Comment by Arnavion** [comment]
>If there is duplication the definitions should be duplicated

Okay, changed it to redeclare `DISPID` and `MEMBERID` instead of importing the latter from `oaidl`

File: src/um/oleauto.rs (hunk 3 above)
Comment (Arnavion) [inline src/um/oleauto.rs]: Fixed.

File: src/um/oaidl.rs (hunk 2 above)
Comment (Arnavion) [inline src/um/oaidl.rs]: Fixed.

**Pull request merged**